Makefile

```diff
@@ -163,7 +163,7 @@ clean: ## Clean.
 export KUBEBUILDER_ASSETS=$(PROJECT_DIR)/bin
 
 .PHONY: test
-test: lint generate-deepcopy generate-mocks bin/ginkgo bin/kubectl bin/kube-apiserver bin/etcd ## Run tests. At the moment this is only unit tests.
+test: generate-mocks lint generate-deepcopy bin/ginkgo bin/kubectl bin/kube-apiserver bin/etcd ## Run tests. At the moment this is only unit tests.
```

Comment:
If the mocks need updating, they need to be updated prior to linting and other commands.

Reply:
Why is this ordering critical, particularly wrt lint, which I imagine to be read-only.

Reply:
The mocks are used in the tests. If the mock clients don't match the interface anymore as the actual has changed then `go fmt` and `go vet` break as the test code just wouldn't even compile.